pkg/database/openshiftclusters.go

```diff
@@ -284,8 +284,14 @@ func (c *openShiftClusters) EndLease(ctx context.Context, key string, provisioni
 		if provisioningState == api.ProvisioningStateSucceeded {
 			doc.Dequeues = 0
 		}
-
-		doc.AsyncOperationID = ""
+		// If EndLease is called while cluster is still in Creating phase,
+		// we flush dequeues count and don't clean AsyncOperationID as this is
+		// just handover between phases
+		if provisioningState != api.ProvisioningStateCreating {
+			doc.AsyncOperationID = ""
+		} else {
+			doc.Dequeues = 0
+		}
```

Comment:
given lines 284-286, writing this adds technical debt.

I guess that if the provisioning state is terminal, AsyncOperationID should be reset?
And if the provisioning state is not failed, Dequeues should be reset?

Reply:
At this point dequeue behaviour and intention is not clear for me. If points is to catch errors - yes, we should reset it in the non failed state.
But if we have behaviour, where Cluster is stuck in Creating phase for past 8 dequeue because we have a logical bug somewhere, we will not catch it as dequeue will get zeroed, and we risk in overloading backends.

If we say "dequeue limit is 5 + number of phases" we assume that cluster should finish in 5 dequeues, if not - fail, and move on. 

I somehow think that we should reset this counter only when we finish, and keep it as threshold safe switch.

Reply:
> If we say "dequeue limit is 5 + number of phases" 

For now I'd rather avoid coupling those two things.  I'm OK with zeroing Dequeues if we think we succeeded or we finished a phase OK - in the case that we crash and drop the lease, or we returned an error, we will not zero Dequeues.